src/main/java/org/assertj/core/internal/Diff.java

```diff
@@ -66,11 +67,29 @@
     return diff(actual.toPath(), expected, charset);
   }
 
+  @VisibleForTesting
+  public List<Delta<String>> diff(Object[] actual, Object[] expected) throws IOException {
+    return diff(
+      readerFor(arrayToStringWithDelimeter(actual, org.assertj.core.util.Compatibility.System.lineSeparator())),
```

Comment:
extract `org.assertj.core.util.Compatibility.System.lineSeparator()` to a local variable for readability